tracer-core/src/test/java/org/zalando/tracer/GeneratorTest.java

```diff
@@ -41,4 +46,26 @@ public void testFlowId() throws Exception {
         assertThat(value.length(), is(22));
     }
 
-}
\ No newline at end of file
+    @Test
+    public void testPhrase() throws Exception {
+        final String value = new PhraseGenerator().generate();
+
+        assertThat(value, matchesPattern("[a-z]+_[a-z]+_[a-z]+_[a-z]+_[a-z]+"));
+
+        assertTrue(PhraseGenerator.isJacocoHappy());
+    }
+
+    @Test
+    public void testPhraseWozniakIsNotBoring() throws Exception {
+        final Iterator<Integer> indexes = Arrays.asList(9, 143, 0, 0, 0, 0, 0, 0, 0, 0).iterator();
+
+        final String value = PhraseGenerator.generate(i -> indexes.next());
```

Comment:
Assuming you simplify the above declaration to a `List<Integer>` should could change this then to:

```java
final String value = PhraseGenerator.generate(indexes::get);
```